pkg/didcomm/protocol/mediator/service.go

```diff
@@ -549,41 +527,69 @@ func (s *Service) doRegistration(record *connection.Record, req *Request, timeou
 		return fmt.Errorf("ensure connection exists: %w", err)
 	}
 
-	// register chan for callback processing
-	grantCh := make(chan Grant)
-	s.setRouteRegistrationCh(req.ID, grantCh)
-
 	// TODO: would this be better served as time.Now().Add(timeout).Unix() as pkg/doc/verifiable/credential.go
 	// demonstrates? additionally `ExpiresTime` would need to be migrated to int64
 	req.ExpiresTime = time.Now().UTC().Add(timeout)
 
 	// send message to the router
-	if err := s.outbound.SendToDID(req, record.MyDID, record.TheirDID); err != nil {
+	if err = s.outbound.SendToDID(req, record.MyDID, record.TheirDID); err != nil {
 		return fmt.Errorf("send route request: %w", err)
 	}
 
-	// callback processing (to make this function look like a sync function)
```

Comment:
Can we have a comment something like this ? It'll help to understand why we are doing this when one someone looks at the code.

Reply:
Done!